internal/store/elasticsearch/discovery_search_repository.go

```diff
@@ -268,6 +285,10 @@ func (repo *DiscoveryRepository) toSearchResults(hits []searchHit) []asset.Searc
 		if id == "" { // this is for backward compatibility for asset without ID
 			id = r.URN
 		}
+
+		if r.Data != nil && hit.HighLight != nil {
+			r.Data["highlight"] = hit.HighLight
```

Comment:
Let's use a field name less likely to conflict, something like `_highlight`/`@highlight`. We should also handle the case when `r.Data` is nil (assign new map to `r.Data`).

Reply:
done